src/openms/source/FILTERING/ID/IDFilter.cpp

```diff
@@ -875,50 +875,55 @@ namespace OpenMS
 
     vector<IdentificationData::ObservationMatchRef> best_matches =
       id_data.getBestMatchPerObservation(score_ref);
+
     auto best_match_it = best_matches.begin();
-    for (auto it = id_data.observation_matches_.begin();
-         it != id_data.observation_matches_.end(); )
-    {
+
+    // predicate to compare the best match(es) to all (ordered) observation matches
+    // returns false if the current om is a best match (-> not not be removed)
+    // returns true if an inferior om was found (-> will be removed)
+    auto is_worse_score = [&](IdentificationData::ObservationMatchRef it)->bool 
```

Comment:
"has_worse_score"? (since you're comparing objects with scores, not the scores directly)

Maybe list `best_match_it` in the capture clause explicitly, to make clear that it's affected.